website/src/panel-layout.tsx

```diff
@@ -0,0 +1,264 @@
+import * as css from "./panel-layout.css"
+import { WordPanel } from "./word-panel"
+
+export interface PanelDetails {
+  currContents: FormFieldsFragment | null
+  setCurrContents: (currContents: FormFieldsFragment | null) => void
+}
+
+/** Displays the right-side panel information of the currently selected word. */
+export const PanelLayout = (props: {
+  segment: FormFieldsFragment | null
+  setContent: (content: FormFieldsFragment | null) => void
+}) => {
+  if (!props.segment) {
+    return null
+  }
+
+  const { form, isEditing } = useForm()
+  const token = useCredentials()
+
+  return (
+    <div className={css.wordPanelContent}>
+      <header className={css.wordPanelHeader}>
+        <div className={css.headerButtons}>
+          <IconButton
+            onClick={() => props.setContent(null)}
+            aria-label="Dismiss selected word information"
+          >
+            <MdClose size={32} />
+          </IconButton>
+
+          {token && <EditButton />}
+        </div>
+
+        <h2 className={css.cherHeader}>{props.segment.source}</h2>
+      </header>
+
+      {/* Renders audio recording. */}
+      <AudioPanel segment={props.segment} />
+
+      {isEditing ? (
+        <Form {...form}>
+          <PanelContent panelType={EditWordPanel} word={props.segment} />
+        </Form>
+      ) : (
+        <PanelContent panelType={WordPanel} word={props.segment} />
+      )}
+    </div>
+  )
+}
+
+/** Dispatches to the corresponding panel type to render a normal word panel or an editable word panel. */
+export const PanelContent = (props: {
+  panelType: typeof EditWordPanel | typeof WordPanel
+  word: FormFieldsFragment
+}) => {
+  // Contains components rendering data of a word's phonetics.
+  const phoneticsContent = (
+    <>
+      {props.word.source && (
+        <props.panelType
+          word={props.word}
+          feature={"source"}
+          label="Syllabary Characters"
+        />
+      )}
+
+      {props.word.simplePhonetics && (
+        <props.panelType
+          word={props.word}
+          feature={"simplePhonetics"}
+          label="Simple Phonetics"
+        />
+      )}
+
+      {props.word.romanizedSource &&
+        props.word.romanizedSource !== props.word.simplePhonetics && (
+          <props.panelType
+            word={props.word}
+            feature={"romanizedSource"}
+            label="Romanized Source"
+          />
+        )}
+    </>
+  )
+
+  const translation = (
+    <>
+      {props.word.englishGloss[0] !== "" && (
+        <props.panelType
+          word={props.word}
+          feature={"englishGloss"}
+          label="English Translation"
+        />
+      )}
+    </>
+  )
+
+  // Contains components rendering a word's segments and its english translation.
+  const wordPartsContent = (
+    <>
+      <VerticalMorphemicSegmentation
+        panelType={props.panelType}
+        segments={props.word.segments}
+      />
+
+      {props.panelType === WordPanel ? (
```

Comment:
Hmm it feels weird to use a component as a type distinguisher here, no? Maybe you need an enum?

Reply:
Since I can't use object literals as values for enums, I have an additional line here now that checks the enum type and then sets the component type that is rendered - I hope this would be okay!

![image](https://user-images.githubusercontent.com/40529597/182208519-32f0ebe9-c20e-4a3a-8e7c-a4b019b1648f.png)

Reply:
Yeah that works, though use the enum variant name rather than a number like `0`